### ISSUE TITLE
Known FFmpeg issues that probably should be reported/fixed upstream

### ISSUE THREAD

**Issue by myrsloik** [vapoursynth/bestsource#87]
1. Some FFmpeg encoders don't flag keyframes. The HuffYUV encoder is the most notable case of this. Every frame is a keyframe so why can't it be marked as such?
2. VC-1 decoder output isn't bitexact after seeking.
3. Somewhat corrupt MPEG2 files such as the one in https://github.com/vapoursynth/bestsource/issues/31 don't produce bitexact output with repeated linear decoding. This probably indicates harmless uninitialized memory somewhere.

**Comment by dartheditous** [comment]
I think bestsource has the same symptom as FFMS2 in that it isn't able to efficiently seek audio in MKV files. It creates an audio index, but if you jump around on the timeline and hit play, there is a delay/freeze which is proportional to the position of the jump (there is no delay when audio isn't being accessed, e.g. either by not sourcing it in the first place or by not trying to play audio (e.g. in VirtualDub, audio isn't accessed unless either the Audio Display open, or you hit play).

With the VirtualDub Audio Display open, clicking on the timeline results in a near-instant change of the video frame, then VirtualDub freezes for a while until it has the audio.

Remuxing to MP4 solves the problem.

Is that likely to be down to FFmpeg?

**Comment by myrsloik** [comment]
Yes, probably has something to do with the demuxer if MP4 is fast. If you send a pair of test files I can tell you exactly what goes wrong if you plan to try to badger FFmpeg developers about it.
